### PR TITLE
Add `hybridLaunchKernel` dispatch

### PR DESCRIPTION
**Pull request by 1tnguyen**
### Description

Dispatch kernel launch based on whether JIT+synthesis would be required (remote platforms). Otherwise, the direct launch function will be used.

### CODE REVIEW

File: runtime/cudaq/platform/qpu.h [hunk 1]
```diff
@@ -174,8 +174,17 @@ class QPU : public registry::RegisteredType<QPU> {
...
+
+  /// Launch the kernel with given name and argument arrays.
+  // This is intended for remote QPUs whereby we need to JIT-compile the kernel
+  // with argument synthesis. Remote QPU implementation to override this.
   virtual void launchKernel(const std::string &name,
-                            const std::vector<void *> &rawArgs) = 0;
+                            const std::vector<void *> &rawArgs) {
+    if (!isRemote())
+      throw std::runtime_error("Wrong kernel launch point: Attempt to launch "
+                               "kernel in JIT mode on local "
```
Comment (schweitzpgi) [inline runtime/cudaq/platform/qpu.h:185]: The documentation calls this "streamlined for JIT". Can we add that phrasing here?

Comment (1tnguyen) [reply runtime/cudaq/platform/qpu.h]: Sure. Fixed in https://github.com/NVIDIA/cuda-quantum/pull/2111/commits/83b67ba456c22704dd11a56b65bb7ca1b2c27ff7

File: include/cudaq/Optimizer/Transforms/Passes.td [hunk 2]
```diff
@@ -332,7 +332,7 @@ def GenerateKernelExecution : Pass<"kernel-execution", "mlir::ModuleOp"> {
       /*default=*/"\"-\"", "Name of output file.">,
     Option<"startingArgIdx", "starting-arg-idx", "std::size_t", /*default=*/"0",
       "The starting argument index for the argsCreator.">,
-    Option<"codegenKind", "codegen", "std::size_t", /*default=*/"1",
+    Option<"codegenKind", "codegen", "std::size_t", /*default=*/"0",
```
[the remark this answers is not in the document]
Comment (1tnguyen) [reply include/cudaq/Optimizer/Transforms/Passes.td:335]: Perhaps I'm lucky but all tests are passing now :)

[the remark this answers is not in the document]
Comment (schweitzpgi) [reply include/cudaq/Optimizer/Transforms/Passes.td:335]: Got through the CI without a wipeout. Nice.

**Review by schweitzpgi — APPROVED** [review]
Thanks, Thien.

File: test/Quake-QIR/return_values.qke [hunk 3]
```diff
@@ -6,7 +6,7 @@
 // the terms of the Apache License 2.0 which accompanies this distribution.   //
 // ========================================================================== //
 
-// RUN: cudaq-opt --add-dealloc --kernel-execution --canonicalize %s | \
+// RUN: cudaq-opt --add-dealloc --kernel-execution=codegen=1 --canonicalize %s | \
```
Comment (schweitzpgi) [inline test/Quake-QIR/return_values.qke:9]: We should add a work item to extend tests like this to check both the `codegen=1` and `codegen=0` outputs. The longer range plan to deprecate the `codegen=1` option.

**Pull request merged**